casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -101,8 +103,9 @@ object BlockAPI {
                          }
               } yield result).timer("propose-total-time").attempt.flatMap {
                 case Left(e) =>
-                  Metrics[F].incrementCounter("propose-failed") >> Sync[F]
-                    .raiseError[ApiErr[DeployServiceResponse]](e)
+                  Metrics[F].incrementCounter("propose-failed") >>
+                    Log[F].error(s"Error while creating block: ${e.getMessage}") >>
```

Comment:
Are you sure we need this line? Won't it be logged twice because of `ProposeGrpcService.defer` (it calls `attemptAndLog` internally)?

Reply:
yes, I know. But `attemptAndLog` actually doesn't log anything
```scala
      case Left(ex)      => Task.delay(ex.printStackTrace()) >> Task.raiseError[A](ex)
```
We can remove this line after we change `attemptAndLog`.